### PR TITLE
Remove ruff config excludes

### PR DESCRIPTION
**Pull request by hillairet**
Fix: #968

So I removed all the ignores of specific errors and the exclude of `loguru/__init__.pyi`

* For `loguru/__init__.pyi` I just added `noqa: N805` (thanks for the detailed explanation @Delgan )
* For [B018](https://beta.ruff.rs/docs/rules/useless-expression/), it was on a bunch of `1 / 0` in the tests so I added `noqa: B018`
* For [B028](https://beta.ruff.rs/docs/rules/no-explicit-stacklevel/), I made added `stacklevel=1` which is the default value but bugbear wants this to be explicit. So no change of behavior expected.
* For [B033](https://beta.ruff.rs/docs/rules/duplicate-value/), it was simply a duplicated entry in a set in `tests/test_filesink_retention.py` so I removed the duplicate.
* :warning: For [B904](https://beta.ruff.rs/docs/rules/raise-without-from-inside-except/), I added a `from` clause because the code was reraising an exception without it. That one needs your review @Delgan!

### CODE REVIEW

File: loguru/_logger.py [hunk 1]
```diff
@@ -890,11 +890,11 @@ def add(
                 elif isinstance(level_, str):
                     try:
                         levelno_ = self.level(level_).no
-                    except ValueError:
+                    except ValueError as err:
                         raise ValueError(
                             "The filter dict contains a module '%s' associated to a level name "
                             "which does not exist: '%s'" % (module, level_)
-                        )
+                        ) from err
```
Comment (Delgan) [inline loguru/_logger.py:897]: Can we just `raise` from `None` here please? It's consistent with other raised `Exception` in the module. I prefer to hide implementation details from the stack traceback.

File: loguru/_logger.py [hunk 2]
```diff
@@ -1378,6 +1378,7 @@ def opt(
             warnings.warn(
                 "The 'ansi' parameter is deprecated, please use 'colors' instead",
                 DeprecationWarning,
+                stacklevel=1,
```
Comment (Delgan) [inline loguru/_logger.py:1381]: Given than [`B028`](https://beta.ruff.rs/docs/rules/no-explicit-stacklevel/) considers the default `stacklevel=1` as bad, shouldn't we replace it with `stacklevel=2`?

It seems indeed best practice, I didn't realize that the warnings weren't even visible to the user by default...

It slightly changes the behavior but this is for the better in my opinion.

File: loguru/_logger.py (hunk 1 above)
Comment (hillairet) [reply loguru/_logger.py]: Makes sense. Done.

File: loguru/_logger.py (hunk 2 above)
Comment (hillairet) [reply loguru/_logger.py]: Sure thing. Done.

**Comment by Delgan** [comment]
This is perfect, I'm really glad to see all these code smells addressed!
Thanks a lot for your work and all the details. :+1:

**Pull request merged**